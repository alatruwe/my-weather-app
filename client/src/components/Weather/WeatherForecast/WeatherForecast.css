.weatherForecast {
  display: flex;
  justify-content: space-evenly;
}

.weatherDetails {
  display: flex;
  align-items: center;
}

.weatherDetailsIcon {
  margin-right: 5%;
}
