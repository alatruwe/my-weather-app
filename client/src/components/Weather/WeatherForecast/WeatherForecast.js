import { Component } from "react";
import data from "../../../data.json";
import WeatherCard from "../WeatherCard/WeatherCard";
import "./WeatherForecast.css";

class WeatherForecast extends Component {
  // get data from context here
  // pass down data as props to WeatherCard
  render() {
    return (
      <div className="wrapperForecast">
        <ul className="weatherForecast">
          {data.list.slice(1).map((data, index) => (
            <li key={data.dt}>
              <h2>Day {index + 2}</h2>

              <WeatherCard
                icon={data.weather[0].icon}
                description={data.weather[0].description}
                temp={data.main.temp}
                min={data.main.temp_min}
                max={data.main.temp_max}
                feelsLike={data.main.feels_like}
                humidity={data.main.humidity}
                wind={data.wind.speed}
              />
            </li>
          ))}
        </ul>
      </div>
    );
  }
}

export default WeatherForecast;
