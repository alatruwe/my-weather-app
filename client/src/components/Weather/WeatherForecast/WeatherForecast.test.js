import React from "react";
import ReactDOM from "react-dom";
import WeatherForecast from "./WeatherForecast";
import renderer from "react-test-renderer";

describe("WeatherCity", () => {
  it("renders without crashing", () => {
    const div = document.createElement("div");
    ReactDOM.render(<WeatherForecast />, div);
    // clean up code
    ReactDOM.unmountComponentAtNode(div);
  });

  it("renders the UI as expected", () => {
    const tree = renderer.create(<WeatherForecast />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
