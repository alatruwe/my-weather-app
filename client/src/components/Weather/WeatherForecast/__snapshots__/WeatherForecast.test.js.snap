// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WeatherCity renders the UI as expected 1`] = `
<div
  className="wrapperForecast"
>
  <ul
    className="weatherForecast"
  >
    <li>
      <h2>
        Day 
        2
      </h2>
      <div
        className="card"
        onClick={[Function]}
      >
        <div
          className="vertical-border"
        />
        <div
          className="cardDetails"
        >
          <div>
            <img
              alt="weather icon"
              src="../../../assets/art_clear.png"
            />
            <h2>
              clear sky
            </h2>
          </div>
          <div
            className="weatherDetails"
          >
            <div
              className="weatherDetailsIcon"
            >
              <i
                className="fas fa-thermometer-half"
              />
              <p>
                Temp
              </p>
            </div>
            <p>
              83.5
               F
            </p>
          </div>
          <div
            className="weatherDetails"
          >
            <div
              className="weatherDetailsIcon"
            >
              <i
                className="fas fa-sun"
              />
              <p>
                UV index
              </p>
            </div>
            <p>
              UV index
            </p>
          </div>
          <div />
        </div>
      </div>
    </li>
    <li>
      <h2>
        Day 
        3
      </h2>
      <div
        className="card"
        onClick={[Function]}
      >
        <div
          className="vertical-border"
        />
        <div
          className="cardDetails"
        >
          <div>
            <img
              alt="weather icon"
              src="../../../assets/art_clouds.png"
            />
            <h2>
              scattered clouds
            </h2>
          </div>
          <div
            className="weatherDetails"
          >
            <div
              className="weatherDetailsIcon"
            >
              <i
                className="fas fa-thermometer-half"
              />
              <p>
                Temp
              </p>
            </div>
            <p>
              84.94
               F
            </p>
          </div>
          <div
            className="weatherDetails"
          >
            <div
              className="weatherDetailsIcon"
            >
              <i
                className="fas fa-sun"
              />
              <p>
                UV index
              </p>
            </div>
            <p>
              UV index
            </p>
          </div>
          <div />
        </div>
      </div>
    </li>
    <li>
      <h2>
        Day 
        4
      </h2>
      <div
        className="card"
        onClick={[Function]}
      >
        <div
          className="vertical-border"
        />
        <div
          className="cardDetails"
        >
          <div>
            <img
              alt="weather icon"
              src="../../../assets/art_clear.png"
            />
            <h2>
              clear sky
            </h2>
          </div>
          <div
            className="weatherDetails"
          >
            <div
              className="weatherDetailsIcon"
            >
              <i
                className="fas fa-thermometer-half"
              />
              <p>
                Temp
              </p>
            </div>
            <p>
              85.08
               F
            </p>
          </div>
          <div
            className="weatherDetails"
          >
            <div
              className="weatherDetailsIcon"
            >
              <i
                className="fas fa-sun"
              />
              <p>
                UV index
              </p>
            </div>
            <p>
              UV index
            </p>
          </div>
          <div />
        </div>
      </div>
    </li>
    <li>
      <h2>
        Day 
        5
      </h2>
      <div
        className="card"
        onClick={[Function]}
      >
        <div
          className="vertical-border"
        />
        <div
          className="cardDetails"
        >
          <div>
            <img
              alt="weather icon"
              src="../../../assets/art_clear.png"
            />
            <h2>
              clear sky
            </h2>
          </div>
          <div
            className="weatherDetails"
          >
            <div
              className="weatherDetailsIcon"
            >
              <i
                className="fas fa-thermometer-half"
              />
              <p>
                Temp
              </p>
            </div>
            <p>
              89.22
               F
            </p>
          </div>
          <div
            className="weatherDetails"
          >
            <div
              className="weatherDetailsIcon"
            >
              <i
                className="fas fa-sun"
              />
              <p>
                UV index
              </p>
            </div>
            <p>
              UV index
            </p>
          </div>
          <div />
        </div>
      </div>
    </li>
  </ul>
</div>
`;
